creator-node/src/services/stateMachineManager/utils.js

```diff
@@ -0,0 +1,23 @@
+class StateMachineUtils {
+  /**
+   * @param {Object[]} nodeUserInfoList array of objects of schema { primary, secondary1, secondary2, user_id, wallet }
+   * @returns {Set} Set of content node endpoint strings
+   */
+  static computeContentNodePeerSet(nodeUserInfoList, thisContentNodeEndpoint) {
+    // Aggregate all nodes from user replica sets
+    let peerList = nodeUserInfoList
+      .map((userInfo) => userInfo.primary)
+      .concat(nodeUserInfoList.map((userInfo) => userInfo.secondary1))
+      .concat(nodeUserInfoList.map((userInfo) => userInfo.secondary2))
+
+    peerList = peerList
+      .filter(Boolean) // filter out false-y values to account for incomplete replica sets
+      .filter((peer) => peer !== thisContentNodeEndpoint) // remove self from peerList
+
+    const peerSet = new Set(peerList) // convert to Set to get uniques
+
+    return peerSet
+  }
+}
```

Comment:
only change is making this endpoint be a param (`thisContentNodeEndpoint`) so it doesn't need state and can be a pure util function instead. see: https://github.com/AudiusProject/audius-protocol/blob/4c559fd7b2a7e3aa5e4772c5b153260841fe9072/creator-node/src/snapbackSM/peerSetManager.js#L206-L224